File: api/app/Module/Validation/Log/Request.php
<?php

declare(strict_types=1);

namespace App\Module\Validation\Log;

use App\Module\Validation\AbstractValidation;

class Request extends AbstractValidation
{
    protected array $rule = [
        'logId' => 'sometimes|required|integer|min:1',
        'requestUrl' => 'sometimes|required|url',
        'minRunTime' => 'sometimes|required|numeric|min:0',
        'maxRunTime' => 'sometimes|required|numeric|min:0|gte:minRunTime',
    ];

    protected array $scene = [];
}

File: api_go/api/log/request.go
package api

import (
	apiCommon "api/api"
)

type RequestListReq struct {
	apiCommon.CommonListReq
	Filter RequestListFilterReq `p:"filter"`
}

type RequestListFilterReq struct {
	apiCommon.CommonListFilterReq `c:",omitempty"`
	LogId                         *uint    `c:"logId,omitempty" p:"logId" v:"integer|min:1"`
	RequestUrl                    string   `c:"requestUrl,omitempty" p:"requestUrl" v:"url"`
	MinRunTime                    *float64 `c:"minRunTime,omitempty" p:"minRunTime" v:"float|min:0"`
	MaxRunTime                    *float64 `c:"maxRunTime,omitempty" p:"maxRunTime" v:"float|min:0|gte:MinRunTime"`
}
